src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/handlers/presubmit/CaseListedCaseOfficerNotifier.java

```diff
@@ -0,0 +1,125 @@
+
+@Component
+public class CaseListedCaseOfficerNotifier implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final String caseListedCaseOfficerTemplateId;
+    private final CaseOfficerPersonalisationFactory caseOfficerPersonalisationFactory;
+    private final Map<HearingCentre, String> hearingCentreEmailAddresses;
+    private final NotificationSender notificationSender;
+    private final NotificationIdAppender notificationIdAppender;
+    private final StringProvider stringProvider;
+
+    public CaseListedCaseOfficerNotifier(
+        @Value("${govnotify.template.caseListedCaseOfficer}") String caseListedCaseOfficerTemplateId,
+        CaseOfficerPersonalisationFactory caseOfficerPersonalisationFactory,
+        Map<HearingCentre, String> hearingCentreEmailAddresses,
+        NotificationSender notificationSender,
+        NotificationIdAppender notificationIdAppender,
+        StringProvider stringProvider
+    ) {
+        requireNonNull(caseListedCaseOfficerTemplateId, "caseListedCaseOfficerTemplateId must not be null");
+
+        this.caseListedCaseOfficerTemplateId = caseListedCaseOfficerTemplateId;
+        this.caseOfficerPersonalisationFactory = caseOfficerPersonalisationFactory;
+        this.hearingCentreEmailAddresses = hearingCentreEmailAddresses;
+        this.notificationSender = notificationSender;
+        this.notificationIdAppender = notificationIdAppender;
+        this.stringProvider = stringProvider;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && callback.getEvent() == Event.LIST_CASE;
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        final HearingCentre listedHearingCentre =
+            asylumCase
+                .read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)
+                .orElseThrow(() -> new IllegalStateException("listCaseHearingCentre is not present"));
+
+        final String hearingCentreEmailAddress =
+            hearingCentreEmailAddresses
+                .get(listedHearingCentre);
+
+        if (hearingCentreEmailAddress == null) {
+            throw new IllegalStateException("Hearing centre email address not found: " + listedHearingCentre.toString());
+        }
+
+        final String hearingCentreAddress =
```

Comment:
Not really using this value anywhere here.  Remove or pass it to the personalisationFactory.